pylint/checkers/typecheck.py

```diff
@@ -1153,43 +1151,67 @@ def _check_assignment_from_function_call(self, node):
         if not isinstance(function_node, funcs):
             return
 
-        # Unwrap to get the actual function object
+        # Unwrap to get the actual function node object
         if isinstance(function_node, astroid.BoundMethod) and isinstance(
             function_node._proxied, astroid.UnboundMethod
         ):
             function_node = function_node._proxied._proxied
 
         # Make sure that it's a valid function that we can analyze.
         # Ordered from less expensive to more expensive checks.
-        # pylint: disable=too-many-boolean-expressions
         if (
             not function_node.is_function
-            or isinstance(function_node, nodes.AsyncFunctionDef)
-            or function_node.decorators
-            or function_node.is_generator()
-            or function_node.is_abstract(pass_is_abstract=False)
-            or utils.is_error(function_node)
-            or not function_node.root().fully_defined()
+            or function_node.decorators is not None
+            or self._is_ignored_function(function_node)
         ):
             return
 
-        returns = list(
+        # Fix a false-negative for list.sort(), see issue #5722
+        if self._list_sort_method(node.value):
+            self.add_message("assignment-from-none", node=node)
+            return
+
+        if not function_node.root().fully_defined():
+            return
+
+        return_nodes = list(
             function_node.nodes_of_class(nodes.Return, skip_klass=nodes.FunctionDef)
         )
-        if not returns:
+        if not return_nodes:
             self.add_message("assignment-from-no-return", node=node)
         else:
-            for rnode in returns:
+            for ret_node in return_nodes:
                 if not (
-                    isinstance(rnode.value, nodes.Const)
-                    and rnode.value.value is None
-                    or rnode.value is None
+                    isinstance(ret_node.value, nodes.Const)
+                    and ret_node.value.value is None
+                    or ret_node.value is None
                 ):
                     break
             else:
                 self.add_message("assignment-from-none", node=node)
 
-    def _check_dundername_is_string(self, node):
+    @staticmethod
+    def _is_ignored_function(
+        function_node: Union[
+            nodes.FunctionDef, astroid.UnboundMethod, astroid.BoundMethod
+        ]
+    ) -> bool:
+        return (
+            function_node.is_generator()
+            or function_node.is_abstract(pass_is_abstract=False)
+            or isinstance(function_node, nodes.AsyncFunctionDef)
+            or utils.is_error(function_node)
```

Comment:
```suggestion
            isinstance(function_node, nodes.AsyncFunctionDef)
            or utils.is_error(function_node)
            or function_node.is_generator()
            or function_node.is_abstract(pass_is_abstract=False)
```

I believe this is the order of the most expensive function calls (least>most)

Reply:
@DanielNoord  okay. I just tried to maintain the order it was because in the comment they left in the code they say it was already ordered this way...

Reply:
Ah okay. I looked at all the implementation, I think this should be slightly better.